### ISSUE TITLE
Error while Triggering Replicate Single Object on ....

### ISSUE THREAD

**Issue by th0u** [zjorz/Public-AD-Scripts#6]
Hello,
during our preparation for the krbtgt password change, we run the tests using the script Reset-KrbTgt-Password-For-RWDCs-And-RODCs.ps1 V2.8.
Test#2 (Simulation) on Scope #1 (all RWDC) ran without issues over our 5 RWDCs.

But Test#2 on Scope #3 fails on about 27 of 32 RODCs with the message 
```
[2022-03-15 17:32:49] : Exception Type......: System.Management.Automation.MethodInvocationException
[2022-03-15 17:32:49] : 
[2022-03-15 17:32:49] : Exception Message...: Exception calling "SetInfo" with "0" argument(s): "An operations error occurred.
"
[2022-03-15 17:32:49] : 
[2022-03-15 17:32:49] : Error On Script Line: 797
[2022-03-15 17:32:49] : 

```

Logfile Excerpt:
```

[2022-03-15 17:32:48] :   --> Chosen: continue
[2022-03-15 17:32:48] : 
[2022-03-15 17:32:48] : +++++
[2022-03-15 17:32:48] : +++ Processing KrbTgt Account....: 'krbtgt_12345' | 'CN=krbtgt_12345,CN=Users,DC=ourdomain,DC=net' +++
[2022-03-15 17:32:48] : +++ Used By RODC.................: 'ABC-DC.ourdomain.net' (Site: Branch) +++
[2022-03-15 17:32:48] : +++++
[2022-03-15 17:32:48] : 
[2022-03-15 17:32:48] :   --> RWDC To Create Object On..............: 'our-pdc.ourdomain.net'
[2022-03-15 17:32:48] :   --> Full Name Temp Canary Object..........: '_adReplTempObject_krbtgt_12345_20220315173132'
[2022-03-15 17:32:48] :   --> Description...........................: '...!!!.TEMP OBJECT TO CHECK AD REPLICATION IMPACT.!!!...'
[2022-03-15 17:32:48] :   --> Container For Temp Canary Object......: 'CN=Users,DC=ourdomain,DC=net'
[2022-03-15 17:32:48] : 
[2022-03-15 17:32:48] :   --> Temp Canary Object [CN=_adReplTempObject_krbtgt_12345_20220315173132,CN=Users,DC=ourdomain,DC=net] CREATED on RWDC [our-pdc.ourdomain.net]!...
[2022-03-15 17:32:48] : 
[2022-03-15 17:32:48] : 
[2022-03-15 17:32:48] :   =================================================================== CHECK 1 ===================================================================
[2022-03-15 17:32:48] : 
[2022-03-15 17:32:48] :   - Contacting DC in AD domain ...[our-pdc.ourdomain.NET]...(SOURCE RWDC)
[2022-03-15 17:32:48] :      * DC is Reachable...
[2022-03-15 17:32:48] :      * Object [CN=_adReplTempObject_krbtgt_12345_20220315173132,CN=Users,DC=ourdomain,DC=net] exists in the AD database
[2022-03-15 17:32:48] : 
[2022-03-15 17:32:48] :   - Contacting DC in AD domain ...[ABC-DC.ourdomain.NET]...
[2022-03-15 17:32:48] :      * DC is Reachable...
[2022-03-15 17:32:49] : 
[2022-03-15 17:32:49] : Triggering Replicate Single Object On 'ABC-DC.ourdomain.net' From 'CN=NTDS Settings,CN=Backup-DC,CN=Servers,CN=Headquarter,CN=Sites,CN=Configuration,DC=ourdomain,DC=net' Failed For Object 'CN=_adReplTempObject_krbtgt_12345_20220315173132,CN=Users,DC=ourdomain,DC=net' Using The 'Full' Scope...
[2022-03-15 17:32:49] : 
[2022-03-15 17:32:49] : Exception Type......: System.Management.Automation.MethodInvocationException
[2022-03-15 17:32:49] : 
[2022-03-15 17:32:49] : Exception Message...: Exception calling "SetInfo" with "0" argument(s): "An operations error occurred.
"
[2022-03-15 17:32:49] : 
[2022-03-15 17:32:49] : Error On Script Line: 797
[2022-03-15 17:32:49] : 
[2022-03-15 17:32:49] :      * Object [CN=_adReplTempObject_krbtgt_12345_20220315173132,CN=Users,DC=ourdomain,DC=net] now does exist in the AD database
[2022-03-15 17:32:49] : 
[2022-03-15 17:32:49] : 
[2022-03-15 17:32:49] :   --> Start Time......: 2022-03-15 17:32:48
[2022-03-15 17:32:49] :   --> End Time........: 2022-03-15 17:32:49
[2022-03-15 17:32:49] :   --> Duration........: 0,47 Seconds
[2022-03-15 17:32:49] : 
[2022-03-15 17:32:49] :   --> Temp Canary Object [CN=_adReplTempObject_krbtgt_12345_20220315173132,CN=Users,DC=ourdomain,DC=net] DELETED on RWDC [our-pdc.ourdomain.net]!...
[2022-03-15 17:32:49] : 
[2022-03-15 17:32:49] : 
[2022-03-15 17:32:49] : List Of DCs In AD Domain 'ourdomain.net' And Their Timing...
[2022-03-15 17:32:49] : 
[2022-03-15 17:32:49] : 
Host Name               PDC  Site Name    DS Type    IP Address    Reachable Source RWDC FQDN   Time
---------               ---  ---------    -------    ----------    --------- ----------------   ----
our-pdc.ourdomain.net  True  Headquarter  Read/Write 10.10.1.14       True N.A.                  0
ABC-DC.ourdomain.net   False Branch       Read-Only  10.10.13.20      True our-pdc.ourdomain.net 0,45

```

This happens to about 27 of 32 RODC and - interestingly - after repeating  Test#2 scoped on a random "good" RODC again, the formerly "good" RODC failed again (we tested multiple times) also using Master Domain Administrator.

And ideas on this?
Thanks in advance, T

**Comment by zjorz** [comment]
Are you able to tell me more about your scenario? Local or remote forest, without or without a trust, firewalls in between, routing issues, IP connectivity? and anything else you can think of
What's the difference between a "good" RODC and the RODCs failing?

**Comment by th0u** [comment]
It is a local forest. There is a trust to other Domain, but was not in scope. IP Connectivity is plain forward w/o obstacles via MPLS. The domain is quite old but we are on a recent domain level. Most of the RODCs were RWDCs and were demoted to RODC. We tested on workstation and on one RWDC. Tomorrow we will look into eventlogs and try to find any hints.

One thing that puzzled me was that it seems there were different RWDC used within the script. I think the test object was created on "our-PDC" and the replicate trigger seems to be happen on backup-dc (where the created Object was perhaps not yet present).

Would it be possible for you to take a look at the full log? Ideally it would be great if I could hand it over via closed channel.

**Comment by zjorz** [comment]
please contact me through scripts.gallery@iamtec.eu

**Comment by th0u** [comment]
It seems that this is related to a runtime issue in script/AD and may be due to special circumstances
in our domain.
I found that the error message is thrown when the script creates the Temp-Object on RWDC-A and the RODC is set up to replicate from RWDC-B. In this case the Tempo-object is on RWDC-A but not yet on RWDC-B and then the triggered Replication fails.

I did some patching in the script and hardcoded some stuff for our Domain. The patch takes care that the TempObject is created on same RWDC from which the RODC is setup to replicate. Now Mode#2 - Scope#3 runs without errors.

Next step is to validate the remaining Test Modes with the patches.
